src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommandRunner.cs

```diff
@@ -22,147 +23,162 @@ internal class ListPackageCommandRunner : IListPackageCommandRunner
+                                WarnForHttpSources(listPackageArgs, projectModel);
+                                await GetRegistrationMetadataAsync(packages, listPackageArgs);
+                                await AddLatestVersionsAsync(packages, listPackageArgs);
                             }
-                            else
-                            {
-                                if (listPackageArgs.ReportType != ReportType.Default)  // generic list package is offline -- no server lookups
-                                {
-                                    PopulateSourceRepositoryCache(listPackageArgs);
-                                    WarnForHttpSources(listPackageArgs);
-                                    await GetRegistrationMetadataAsync(packages, listPackageArgs);
-                                    await AddLatestVersionsAsync(packages, listPackageArgs);
-                                }
 
-                                bool printPackages = FilterPackages(packages, listPackageArgs);
+                            bool printPackages = FilterPackages(packages, listPackageArgs);
 
-                                // Filter packages for dedicated reports, inform user if none
-                                if (listPackageArgs.ReportType != ReportType.Default && !printPackages)
+                            // Filter packages for dedicated reports, inform user if none
+                            if (listPackageArgs.ReportType != ReportType.Default && !printPackages)
+                            {
+                                switch (listPackageArgs.ReportType)
                                 {
-                                    switch (listPackageArgs.ReportType)
-                                    {
-                                        case ReportType.Outdated:
-                                            Console.WriteLine(string.Format(CultureInfo.CurrentCulture, Strings.ListPkg_NoUpdatesForProject, projectName));
-                                            break;
-                                        case ReportType.Deprecated:
-                                            Console.WriteLine(string.Format(CultureInfo.CurrentCulture, Strings.ListPkg_NoDeprecatedPackagesForProject, projectName));
-                                            break;
-                                        case ReportType.Vulnerable:
-                                            Console.WriteLine(string.Format(CultureInfo.CurrentCulture, Strings.ListPkg_NoVulnerablePackagesForProject, projectName));
-                                            break;
-                                    }
+                                    case ReportType.Outdated:
+                                        projectModel.AddProjectInformation(string.Format(CultureInfo.CurrentCulture, Strings.ListPkg_NoUpdatesForProject, projectModel.ProjectName), ProblemType.Information);
```

Comment:
Similarly to the above. 

This is already represented in the model and as such does not to be specially logged.

The console output renderer can handle it.

Reply:
Yes, console output renderer can handle it, but I feel this pattern is easier to maintain. See other [comment](https://github.com/NuGet/NuGet.Client/pull/4855/files#r1009933275).

Reply:
I think my argument is that it's not really a pattern. It's mix of a few different things.

Reply:
Sure, removed it.